pkg/blobinfocache/default_test.go

```diff
@@ -147,6 +148,12 @@ func TestDefaultCache(t *testing.T) {
 	}()
 	os.Unsetenv("HOME")
 	os.Unsetenv("XDG_DATA_HOME")
+	os.Setenv("_CONTAINERS_ROOTLESS_UID", "-1")
+	cap, err := capability.NewPid(0)
+	require.NoError(t, err)
+	cap.Unset(capability.EFFECTIVE|capability.PERMITTED|capability.INHERITABLE|capability.BOUNDING|capability.AMBIENT, capability.CAP_DAC_OVERRIDE)
+	err = cap.Apply(capability.EFFECTIVE | capability.PERMITTED | capability.INHERITABLE | capability.BOUNDING | capability.AMBIENT)
+	require.NoError(t, err)
```

Comment:
How does this work for non-Linux systems?

With `_CONTAINERS_ROOTLESS_UID=-1` and both `XDG_DATA_HOME` and `HOME` unset, AFAICS `blobInfoCacheDir` should fail and then capabilities don’t matter; am I missing something?

(I suppose it would be natural and reasonably portable to just skip this test if EUID is 0, but now with `_CONTAINERS_ROOTLESS_UID`, this seems to actually trigger exactly the failure that ordinary unprivileged users trigger.)